spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfigurationTests.java

```diff
@@ -68,8 +71,48 @@ public void should_apply_in_memory_metrics_when_meter_registry_class_missing() {
 				});
 	}
 
+	/**
+	 * Duplicates
+	 * {@link org.springframework.cloud.sleuth.sampler.SamplerAutoConfigurationTests}
+	 * intentionally, to ensure configuration condition bugs do not exist.
+	 */
 	@Test
-	public void should_use_B3Propagation_factory_if_no_have_any_config() {
+	public void should_use_NEVER_SAMPLER_when_only_logging() {
```

Comment:
If we have never sampler, only logging but we want the e.g. baggage to be propagated, won't we have a problem?